src/lib.rs

```diff
@@ -206,6 +214,20 @@ impl ToPublicKey for bitcoin::PublicKey {
     }
 }
 
+impl ToPublicKey for bitcoin::schnorr::PublicKey {
+    fn to_public_key(&self) -> bitcoin::PublicKey {
+        // This code should never be used.
+        // But is implemented for completeness
+        let mut data: Vec<u8> = vec![0x02];
+        data.extend(self.serialize().iter());
+        bitcoin::PublicKey::from_slice(&data).expect("Cannot fail")
```

Comment:
IMO `expect("Cannot fail")` is in fact `unwrap()`. The point of the message in `expect` is to provide debugging information to see which specific unreachable statement was executed. Maybe something like `"failed to construct public key from 33-byte slice"` will be more informative.

Overall, I think that we need to introduce `PublicKey::from_bytes` which takes fixed-length array in `rust-bitcoin` and can't error.

Reply:
Yes, indeed. Will add a message "failed to construct 33 Publickey from 0x02 appended x-only key".

> Overall, I think that we need to introduce PublicKey::from_bytes which takes a fixed-length array in rust-bitcoin and can't error.
Not all 33-byte arrays are valid PublicKeys. About slightly less than half of are valid x-only keys. and 0x02 and 0x03 are valid first bytes for 33 byte publickey